internal/generate/servicepackage/endpoint_resolver.go.gtpl

```diff
@@ -6,87 +6,14 @@ import (
 	"context"
 	"fmt"
 	"net"
-{{ if .ClientSDKV1 -}}
-	"net/url"
-{{ end }}
 
-{{- if .ClientSDKV1 }}
-	endpoints_sdkv1 "github.com/aws/aws-sdk-go/aws/endpoints"
-{{- end }}
-{{- if .ClientSDKV2 }}
 	aws_sdkv2 "github.com/aws/aws-sdk-go-v2/aws"
 	{{ .GoV2Package }}_sdkv2 "github.com/aws/aws-sdk-go-v2/service/{{ .GoV2Package }}"
 	smithyendpoints "github.com/aws/smithy-go/endpoints"
-{{- end }}
 	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"github.com/hashicorp/terraform-provider-aws/internal/errs"
 )
 
-{{ if .ClientSDKV1 }}
-var _ endpoints_sdkv1.Resolver = resolverSDKv1{}
-
-type resolverSDKv1 struct {
-	ctx context.Context
-}
-
-func newEndpointResolverSDKv1(ctx context.Context) resolverSDKv1 {
-	return resolverSDKv1{
-		ctx: ctx,
-	}
-}
-
-func (r resolverSDKv1) EndpointFor(service, region string, opts ...func(*endpoints_sdkv1.Options)) (endpoint endpoints_sdkv1.ResolvedEndpoint, err error) {
-	ctx := r.ctx
-
-	var opt endpoints_sdkv1.Options
-	opt.Set(opts...)
-
-	useFIPS := opt.UseFIPSEndpoint == endpoints_sdkv1.FIPSEndpointStateEnabled
-
-	defaultResolver := endpoints_sdkv1.DefaultResolver()
-
-	if useFIPS {
-		ctx = tflog.SetField(ctx, "tf_aws.use_fips", useFIPS)
-
-		endpoint, err = defaultResolver.EndpointFor(service, region, opts...)
-		if err != nil {
-			return endpoint, err
-		}
-
-		tflog.Debug(ctx, "endpoint resolved", map[string]any{
-			"tf_aws.endpoint": endpoint.URL,
-		})
-
-		var endpointURL *url.URL
-		endpointURL, err = url.Parse(endpoint.URL)
-		if err != nil {
-			return endpoint, err
-		}
-
-		hostname := endpointURL.Hostname()
-		_, err = net.LookupHost(hostname)
-		if err != nil {
-			if dnsErr, ok := errs.As[*net.DNSError](err); ok && dnsErr.IsNotFound {
-				tflog.Debug(ctx, "default endpoint host not found, disabling FIPS", map[string]any{
-					"tf_aws.hostname": hostname,
-				})
-				opts = append(opts, func(o *endpoints_sdkv1.Options) {
-					o.UseFIPSEndpoint = endpoints_sdkv1.FIPSEndpointStateDisabled
-				})
-			} else {
-				err = fmt.Errorf("looking up accessanalyzer endpoint %q: %s", hostname, err)
-				return
-			}
-		} else {
-			return endpoint, err
-		}
-	}
-
-	return defaultResolver.EndpointFor(service, region, opts...)
-}
-{{ end }}
-
-{{ if .ClientSDKV2 }}
 var _ {{ .GoV2Package }}_sdkv2.EndpointResolverV2 = resolverSDKv2{}
 
 type resolverSDKv2 struct {
```

Comment:
I'm going with `resolverV2` to reflect the AWS SDK for Go v2 interface type and to avoid potential collisions with types in the non-generated code (I can image `resolver` as a type that a contributor might want to define).